lib/Launcher.js

```diff
@@ -190,6 +191,31 @@ class Launcher {
   }
 }
 
+class ExitEmitter extends EventEmitter {
+  constructor() {
+    super();
+    this.setMaxListeners(Infinity);
+    const processListeners = [ helper.addEventListener(process, 'exit', exitHandler.bind(this)) ];
+    processListeners.push(helper.addEventListener(process, 'SIGINT', sigHandler('SIGINT', 2).bind(this)));
+    processListeners.push(helper.addEventListener(process, 'SIGTERM', sigHandler('SIGTERM', 15).bind(this)));
+    processListeners.push(helper.addEventListener(process, 'SIGHUP', sigHandler('SIGHUP', 1).bind(this)));
+
+    function sigHandler(eventName, exitCode) {
+      return function() {
+        helper.removeEventListeners(processListeners);
+        this.emit(eventName);
+        process.exit(exitCode);
```

Comment:
What if client code has a custom SIGINT handler that doesn't want to process.exit right away?

Reply:
Fair questions. 

1. Isn't this why we allow them to turn it off via [handleSIGINT](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#puppeteerlaunchoptions)?
2. Can they not use [prependListener](https://nodejs.org/api/events.html#events_emitter_prependlistener_eventname_listener) if they want both?